### DISCUSSION TITLE
Support for Dynamic Updates in Clone Operations similar to cp -u

### DISCUSSION THREAD

**Discussion by HavAnIdea** [juicedata/juicefs#4788] [General]
Hello JuiceFS Team,

I've been utilizing JuiceFS for managing my filesystems and have benefited greatly from its features, especially the clone operation for duplicating directories. However, I've encountered a limitation that impacts my workflow significantly.

**Current Behavior:**

JuiceFS currently supports cloning a source directory to a new, non-existent target path. This functionality is great for initial duplication efforts but falls short for ongoing synchronization or update tasks since cloned directories cannot be periodically updated directly via JuiceFS.

**Desired Feature:**

In order to maintain synchronization between the original and cloned directories over time, I find myself resorting to external tools like the cp command with the -u option (copy only when the SOURCE file is newer than the destination file or when the destination file is missing).

What I am seeking is a native capability within JuiceFS that mirrors this cp -u functionality. Ideally, it would be beneficial to have a feature where JuiceFS can not only clone directories but also periodically check for updates in the source directory and apply those changes to the target directory, akin to the behavior of cp -u.

**Use Case:**

This feature is particularly useful for scenarios where continuous data consistency between two directories is required without manually triggering a full clone operation each time, thus ensuring efficiency and saving on unnecessary data transfer operations.

**Suggested Solution:**

Introduce a new flag or command in JuiceFS that allows users to specify that a clone operation should keep the target directory in sync with the source directory by checking for and applying updates at specified intervals.
Alternatively, enhance the existing clone operation with an option to compare the modification dates of files and directories and clone only the differences.
I believe this enhancement will further solidify JuiceFS as a comprehensive solution for filesystem management, catering to a wider range of use cases and improving user experience in maintaining data consistency across directories.

Thank you for considering this feature request. I am looking forward to any feedback or discussion on this proposal.

**Comment by davies** [comment]
clone is to create an isolation version of source, it should NOT follow the changes in source.

If you need a copy that follow the source you can use 1) hardlink for files or 2) symlink for directoires.
